### PR TITLE
Use get_message_body_summary of PSR7 and require min 1.5

### PR DESCRIPTION
**Pull request by gmponos**

### CODE REVIEW

File: src/Exception/RequestException.php [hunk 1]
```diff
@@ -124,32 +124,7 @@ public static function create(
      */
     public static function getResponseBodySummary(ResponseInterface $response)
     {
-        $body = $response->getBody();
-
-        if (!$body->isSeekable() || !$body->isReadable()) {
```
Comment (gmponos) [inline src/Exception/RequestException.php]: Wow... it seems that after I have added the function `get_message_body_summary` which mostly I copied from here... There was this added afterwards https://github.com/guzzle/guzzle/pull/2081 

@sagikazarmark or @Tobion can you release a 1.5.1 containing this change?

**Comment by sagikazarmark** [comment]
Looks like this is in direct conflict with #2272

**Comment by gmponos** [comment]
Why? The function on the PSR-7 package has a parameter on how many chars to use

https://github.com/guzzle/psr7/blob/master/src/functions.php#L853

after merging this the author of the other MR has to resolve conflicts... or it could be the other way around...

File: composer.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
         "php": ">=5.5",
         "ext-json": "*",
         "guzzlehttp/promises": "^1.0",
-        "guzzlehttp/psr7": "^1.4"
+        "guzzlehttp/psr7": "^1.5.2"
```
Comment (Nyholm) [inline composer.json]: This change is not needed. 
Please rebase your PR:

**Comment by lightswitch05** [comment]
Using `guzzlehttp/guzzle:6.5.2`, I'm getting a stack trace around this change: `Fatal error: Call to undefined function GuzzleHttp\Psr7\get_message_body_summary() in /opt/ide/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php on line 127`

```
Fatal error: Call to undefined function GuzzleHttp\Psr7\get_message_body_summary() in /opt/ide/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php on line 127

Call Stack:
    0.0010     247816   1. {main}() /composer/vendor/codeception/codeception/codecept:0
    0.9398    9096216   2. Codeception\Application->run() /composer/vendor/codeception/codeception/codecept:43
    0.9399    9096680   3. Symfony\Component\Console\Application->run() /composer/vendor/codeception/codeception/src/Codeception/Application.php:108
    0.9999    9378256   4. Symfony\Component\Console\Application->doRun() /composer/vendor/symfony/console/Application.php:130
    1.0006    9379528   5. Symfony\Component\Console\Application->doRunCommand() /composer/vendor/symfony/console/Application.php:223
    1.0007    9380128   6. Symfony\Component\Console\Command\Command->run() /composer/vendor/symfony/console/Application.php:869
    1.0076    9388272   7. Codeception\Command\Run->execute() /composer/vendor/symfony/console/Command/Command.php:264
    1.2776   10910368   8. Codeception\Command\Run->runSuites() /composer/vendor/codeception/codeception/src/Codeception/Command/Run.php:387
    1.2776   10910976   9. Codeception\Codecept->run() /composer/vendor/codeception/codeception/src/Codeception/Command/Run.php:492
    1.3235   10959640  10. Codeception\Codecept->runSuite() /composer/vendor/codeception/codeception/src/Codeception/Codecept.php:158
    1.7070   12459992  11. Codeception\SuiteManager->run() /composer/vendor/codeception/codeception/src/Codeception/Codecept.php:191
    1.9569   13342656  12. Codeception\PHPUnit\Runner->doEnhancedRun() /composer/vendor/codeception/codeception/src/Codeception/SuiteManager.php:158
    1.9751   13390816  13. PHPUnit_Framework_TestSuite->run() /composer/vendor/codeception/phpunit-wrapper/src/Runner.php:106
    1.9759   13397080  14. Codeception\Test\Test->run() /composer/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
    2.0280   13422544  15. Codeception\Test\Cest->test() /composer/vendor/codeception/codeception/src/Codeception/Test/Test.php:89
    2.1515   14341760  16. Codeception\Test\Cest->executeTestMethod() /composer/vendor/codeception/codeception/src/Codeception/Test/Cest.php:82
    2.1516   14342152  17. Codeception\Test\Cest->invoke() /composer/vendor/codeception/codeception/src/Codeception/Test/Cest.php:150
    2.1524   14342152  18. Codeception\Lib\Di->injectDependencies() /composer/vendor/codeception/codeception/src/Codeception/Test/Cest.php:138
    2.1527   14343104  19. ReflectionMethod->invokeArgs() /composer/vendor/codeception/codeception/src/Codeception/Lib/Di.php:127
    2.1527   14343352  20. MyClassCest->canDoThing() /composer/vendor/codeception/codeception/src/Codeception/Lib/Di.php:127
    2.1671   14399104  21. Company\App\Client->myFunction() /opt/ide/tests/functional/MyClassCest.php:37
    2.1671   14399760  22. Company\App\Client->executeRequest() /opt/ide/src/Client.php:47
    2.1673   14400704  23. GuzzleHttp\Client->request() /opt/ide/src/Client.php:63
    3.3915   14902376  24. GuzzleHttp\Promise\Promise->wait() /opt/ide/vendor/guzzlehttp/guzzle/src/Client.php:183
    3.3916   14902624  25. GuzzleHttp\Promise\Promise->waitIfPending() /opt/ide/vendor/guzzlehttp/promises/src/Promise.php:62
    3.3916   14902808  26. GuzzleHttp\Promise\Promise->invokeWaitList() /opt/ide/vendor/guzzlehttp/promises/src/Promise.php:225
    3.3917   14902856  27. GuzzleHttp\Promise\Promise->waitIfPending() /opt/ide/vendor/guzzlehttp/promises/src/Promise.php:267
    3.3917   14902976  28. GuzzleHttp\Promise\Promise->invokeWaitFn() /opt/ide/vendor/guzzlehttp/promises/src/Promise.php:223
    3.3918   14903112  29. GuzzleHttp\Promise\TaskQueue->run() /opt/ide/vendor/guzzlehttp/promises/src/Promise.php:246
    3.3924   14901288  30. GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}() /opt/ide/vendor/guzzlehttp/promises/src/TaskQueue.php:47
    3.3925   14901736  31. GuzzleHttp\Promise\Promise::callHandler() /opt/ide/vendor/guzzlehttp/promises/src/Promise.php:156
    3.3926   14901784  32. GuzzleHttp\Middleware::GuzzleHttp\{closure}() /opt/ide/vendor/guzzlehttp/promises/src/Promise.php:203
    3.4150   14946536  33. GuzzleHttp\Exception\RequestException::create() /opt/ide/vendor/guzzlehttp/guzzle/src/Middleware.php:65
    3.4158   14947424  34. GuzzleHttp\Exception\RequestException::getResponseBodySummary() /opt/ide/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:107

```

**Pull request merged**